crates/extensions/c8y_mapper_ext/src/availability/actor.rs

```diff
@@ -0,0 +1,289 @@
+                    self.send_smartrest_set_required_availability_for_child_device(source)
+                        .await?;
+                    self.start_heartbeat_timer_if_interval_is_positive(source)
+                        .await?;
+                }
+                RegistrationResult::Update => {
+                    self.start_heartbeat_timer_if_interval_is_positive(source)
+                        .await?;
+                }
+                RegistrationResult::Error(reason) => warn!(reason),
+            },
+            EntityType::Service => {}
+        }
+
+        Ok(())
+    }
+
+    /// Insert a <"device topic ID" - "health entity topic ID" and "external ID"> pair into the map.
+    /// If @health is provided in the registration message, use the value as long as it's valid as entity topic ID.
+    /// If @health is not provided, use the "tedge-agent" service topic ID as default.
+    /// @id is the only source to know the device's external ID. Hence, @id must be provided in the registration message.
+    fn update_device_service_pair(
+        &mut self,
+        registration_message: &EntityRegistrationMessage,
+    ) -> RegistrationResult {
+        let source = &registration_message.topic_id;
+
+        let result = match registration_message.other.get("@health") {
+            None => registration_message
+                .topic_id
+                .default_service_for_device("tedge-agent")
+                .ok_or_else( || format!("The entity is not in the default topic scheme. Please specify '@health' to enable availability monitoring for the device '{source}'")),
+            Some(raw_value) => match raw_value.as_str() {
+                None => Err(format!("'@health' must hold a string value. Given: {raw_value:?}, source: {source}")),
+                Some(maybe_entity_topic_id) => EntityTopicId::from_str(maybe_entity_topic_id)
+                    .map_err(|_| format!("'@health' must be 4-segment identifier like `a/b/c/d`. Given: {maybe_entity_topic_id}, source: {source}"))
+            }
+        };
+
+        match result {
+            Ok(health_topic_id) => {
+                match registration_message.external_id.clone() {
+                    None => RegistrationResult::Error(format!("'@id' field is missing. Cannot start availability monitoring for the device '{source}'")),
+                    Some(external_id) => {
+                        match self.device_ids_map
+                            .insert(source.clone(), DeviceIds { health_topic_id, external_id }) {
+                            None => RegistrationResult::New,
+                            Some(_) => RegistrationResult::Update,
+                        }
+                    }
+                }
+            }
+            Err(err) => RegistrationResult::Error(err),
+        }
+    }
+
+    /// Set a new timer for heartbeat if the given interval is positive value
+    async fn start_heartbeat_timer_if_interval_is_positive(
```

Comment:
It's not a good idea to name a function after a internal details

```suggestion
    async fn start_heartbeat_timer(
```

Reply:
done in d1c4656